iroh-resolver/src/racing.rs

```diff
@@ -1,36 +1,44 @@
-//! A content loader implementation for iroh-one.
-
+use crate::resolver::{
+    parse_links, ContentLoader, ContextId, LoadedCid, LoaderContext, Source, IROH_STORE,
+};
 use anyhow::{anyhow, Result};
 use async_trait::async_trait;
 use bytes::Bytes;
 use cid::{multibase, Cid};
 use futures::{future::FutureExt, pin_mut, select};
-use iroh_resolver::resolver::{
-    parse_links, ContentLoader, ContextId, LoadedCid, LoaderContext, Source, IROH_STORE,
-};
 use iroh_rpc_client::Client as RpcClient;
+use rand::seq::SliceRandom;
 use tracing::{debug, error, trace, warn};
 
 #[derive(Clone, Debug)]
 pub struct RacingLoader {
     rpc_client: RpcClient,
-    gateway: Option<String>,
+    http_resolvers: Vec<String>,
 }
 
 impl RacingLoader {
-    pub fn new(rpc_client: RpcClient, gateway: Option<String>) -> Self {
+    pub fn new(rpc_client: RpcClient, http_resolvers: Vec<String>) -> Self {
         Self {
             rpc_client,
-            gateway,
+            http_resolvers,
         }
     }
 }
 
 impl RacingLoader {
     pub fn try_raw_gateway(&self) -> Result<&String> {
-        self.gateway
-            .as_ref()
-            .ok_or_else(|| anyhow!("no gateway configured to fetch raw CIDs"))
+        match self.http_resolvers.len() {
+            0 => Err(anyhow!("no gateway configured to fetch raw CIDs")),
+            1 => self
+                .http_resolvers
+                .first()
+                .ok_or_else(|| anyhow!("no gateway configured to fetch raw CIDs")),
```

Comment:
How can this ever happen? We have already checked that there is an element.

Can we just call rand all the time? It's not going to make a difference.

Reply:
> How can this ever happen? We have already checked that there is an element.
yeah this is dumb. Will fix.

> Can we just call rand all the time? It's not going to make a difference.
I was unclear on the overhead of the allocation, and didn't have time to research. I'll trust that you know this is a super low cost &. I'll just call it all the time 😄